pkgs/by-name/fl/flashprog/package.nix

```diff
@@ -24,19 +26,29 @@ stdenv.mkDerivation rec {
     installShellFiles
     pkg-config
   ];
-
   buildInputs = [
     libftdi1
-    libjaylink
     libusb1
   ] ++ lib.optionals (!stdenv.isDarwin) [
-    libgpiod_1
     pciutils
+  ] ++ lib.optionals with_jlink [
+    libjaylink
+  ] ++ lib.optionals (stdenv.isLinux && with_gpio) [
+    libgpiod
   ];
 
-  makeFlags = [ "PREFIX=$(out)" "libinstall" ]
-    ++ lib.optionals stdenv.isDarwin [ "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=no" ]
-    ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [ "CONFIG_INTERNAL_X86=no" "CONFIG_INTERNAL_DMI=no" "CONFIG_RAYER_SPI=0" ];
+  makeFlags = let
+    yesNo = flag: if flag then "yes" else "no";
+  in [
+    "libinstall"
+    "PREFIX=$(out)"
+    "CONFIG_JLINK_SPI=${yesNo with_jlink}"
+    "CONFIG_LINUX_GPIO_SPI=${yesNo (stdenv.isLinux && with_gpio)}"
+    "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=${yesNo (!stdenv.isDarwin)}"
+    "CONFIG_INTERNAL_X86=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
+    "CONFIG_INTERNAL_DMI=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
+    "CONFIG_RAYER_SPI=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
```

Comment:
Aren't the outer () optional?

Reply:
I agree, please remove.

Reply:
No those are not optional. If you remove them you get an syntax error in the listToAttrs builtin: error: syntax error, unexpected '!', expecting '}' This is why the outer () need to stay as the package won't compile otherwise.

Reply:
> Aren't the outer () optional?

I learned this the bad way: ! almost requires ()